aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -344,6 +344,25 @@ mod loader {
             self
         }
 
+        /// Override the name of the app used to build [`SdkConfig`](aws_types::SdkConfig).
+        ///
+        /// This _optional_ name is used to identify the application in the user agent that
+        /// gets sent along with requests.
```

Comment:
Why is it call `AppName` and not `UserAgent`?

Reply:
it's not the `UserAgent` — it eventually gets composed as part of the user agent. But the user agent itself has far more in it than the app name (platform, version, etc.)